### DISCUSSION TITLE
"DefinitelyTyped"-type repository of TypeSpec definitions

### DISCUSSION THREAD

**Discussion by Arithmomaniac** [microsoft/typespec#2846] [Ideas]
Too many public APIs don't publish OpenAPI definitions, and many of those don't even have SDKs. (Two I've wanted recently are [Pocket](https://getpocket.com/developer/) and [Sefaria](https://github.com/Sefaria/Sefaria-Project/wiki/API-Documentation)).
 
I propose a repository of TypeSpec definitions for third-party APIs, just as [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) does for TypeScript. Users could then generate code clients or OpenAPI files for those definitions.

In addition to simply being of help to those who use public APIs on the internet, I imagine the following benefits to the TypeSpec ecosystem:
- Spread awareness of TypeSpec
- Help define patterns and discover potential features/issues when writing TypeSpec files over a wider range of APIs
- Serve as examples of writing TypeSpec files

**Comment by bterlson** [comment]
I think this is a great idea, I have discussed this often with @johanste.  As I recall we had been thinking of it more as a repository for standard types (like GeoJSON, HL7, FIX, whatever) that are standard but for some reason not appropriate to put into our standard library. But I can also see the use case for tabulating specs for APIs in the wild, being able to generate things against popular APIs that don't have typespecs could be very useful.

**Reply by Arithmomaniac** [reply]
This sounds like two different things which could be directories in the same repository:
- Standard(ized) types
- API surfaces (types, methods)
